kimchi/src/circuits/domains.rs

```diff
@@ -21,24 +23,34 @@ impl<F: FftField> EvaluationDomains<F> {
     /// and `d8` (of size `8n`). If generator of `d8` is `g`, the generator
     /// of `d4` is `g^2`, the generator of `d2` is `g^4`, and the generator of `d1` is `g^8`.
     // TODO(mimoo): should we instead panic/return an error if any of these return None?
-    pub fn create(n: usize) -> Option<Self> {
-        let n = Domain::<F>::compute_size_of_domain(n)?;
+    pub fn create(n: usize) -> Result<Self, ProverError> {
+        let n = Domain::<F>::compute_size_of_domain(n)
+            .ok_or(ProverError::Prover("couldn't compute size of domain"))
+            .unwrap();
 
-        let d1 = Domain::<F>::new(n)?;
+        let d1 = Domain::<F>::new(n)
+            .ok_or(ProverError::Prover("could not construct domain"))
+            .unwrap();
 
         // we also create domains of larger sizes
         // to efficiently operate on polynomials in evaluation form.
         // (in evaluation form, the domain needs to grow as the degree of a polynomial grows)
-        let d2 = Domain::<F>::new(2 * n)?;
-        let d4 = Domain::<F>::new(4 * n)?;
-        let d8 = Domain::<F>::new(8 * n)?;
+        let d2 = Domain::<F>::new(2 * n)
+            .ok_or(ProverError::Prover("could not construct domain"))
+            .unwrap();
+        let d4 = Domain::<F>::new(4 * n)
+            .ok_or(ProverError::Prover("could not construct domain"))
+            .unwrap();
+        let d8 = Domain::<F>::new(8 * n)
+            .ok_or(ProverError::Prover("could not construct domain"))
+            .unwrap();
```

Comment:
same here

Reply:
I would suggest errors like "construction of domain d2 did not work as intended"